### ISSUE TITLE
Timelord hashrate

### ISSUE THREAD

**Issue by just-miner** [madMAx43v3r/mmx-node#177]
Ho do you achieve 25+ mh/s iterations, comparing in google table (Timelord (MH/s)? 
My intel Core x7900 with 20 core getting just 2.6 mh/s with just  used 2 core by default.
MMX-node 0.9.14 / Ubuntu 22.10 / 64Gb RAM / Nvidia 4090

**Comment by mwpastore** [comment]
Your processor is 16 years old.

https://www.intel.com/content/www/us/en/products/sku/31730/intel-core2-extreme-processor-x7900-4m-cache-2-80-ghz-800-mhz-fsb/specifications.html

**Comment by just-miner** [comment]
Core i9 7900x

**Comment by mwpastore** [comment]
Thank you for clarifying. 

The single-core performance of Skylake was pretty good in 2015 but doesn't hold up well against Zen 3 and Golden Cove. Furthermore it's missing some specific instructions found in newer architectures that the TL process can leverage to greatly accelerate its processing.

**Comment by just-miner** [comment]
Yep, with AMD Ryzen 9 5950x at 4900 hz (2 core) have 25mh/s in windows binary and about 23mh/s with linux v0.9.14
It seems linux code need better optimizations.

**Comment by QuintLeo** [comment]
> Core i9 7900x
> 
> > https://www.intel.com/content/www/us/en/products/sku/31730/intel-core2-extreme-processor-x7900-4m-cache-2-80-ghz-800-mhz-fsb/specifications.html
> > it has 2 core, not 10 (20 Threads)

I9 7900x IS in fact 10 cores (20 threads) per Intel.
Or the same as my ancient E5-2680 v2 CPUs of the same generation.
It might only be USING 2 of those cores when running a Timelord, but the other cores are there.

But it's single core performance is horrible compared to anything current, and has ZERO support for SHA extensions - and that lack of SHA extensions is the killer in VDF work (that support showed up a FEW generations later).
